frontend/src/pages/Admin/InventoryManagement.js

```diff
@@ -1,173 +1,407 @@
-    price: "LKR3500",
-    size: "M,XL,XXL",
-    quantity: 100,
-    stock: "In Stock",
-    promoCode: "",
-  },
-  {
-    image: pic3,
-    name: "Premium Jogger Pant",
-    description: "Premium quality jogger for individuals",
-    colors: ["black", "blue"],
-    price: "LKR3200",
-    size: "S,M",
-    quantity: 20,
-    stock: "Running Low",
-    promoCode: "FFX",
-  },
-  {
-    image: pic4,
-    name: "Karma Limited Edition",
-    description: "Premium quality oversized for teenagers",
-    colors: ["white", "black", "blue"],
-    price: "LKR3200",
-    size: "M,L,XL",
-    quantity: 0,
-    stock: "Out of Stock",
-    promoCode: "1FF",
-  },
-  {
-    image: pic5,
-    name: "Ladies Top",
-    description: "Comfy top for ladies",
-    colors: ["white", "black", "blue"],
-    price: "LKR3100",
-    size: "M",
+import React, { useState, useEffect } from "react";
+import "./InventoryManagement.css";
+import Select from "../../components/Select";
+import SearchBar from "../../components/SearchBar";
+import Button from "../../components/Button";
+import Table from "../../components/Table";
+import StockStatus from "../../components/StockStatus";
+import api from "../../utils/api";
+
+const InventoryManagement = () => {
+  const [productData, setProductData] = useState([]);
+  const [isFormOpen, setIsFormOpen] = useState(false);
+  const [formMode, setFormMode] = useState("add");
+  const [newProduct, setNewProduct] = useState({
+    image: null,
+    name: "",
+    category: "",
+    description: "",
+    colors: [],
+    price: "",
+    size: "",
     quantity: 0,
-    stock: "Out of Stock",
-    promoCode: "12D",
-  },
-]
-
-//function to return stock status
-const StockStatus = ({ stock }) => {
-  let style
-  switch (stock) {
-    case "In Stock":
-      style = {
-        width: "90px",
-        backgroundColor: "#ffffff",
-        color: "#09db65",
-        padding: "5px",
-        borderRadius: "20px",
-        textAlign: "center",
-        border: "3px , solid ,#09db65",
+    stock: "In Stock",
+  });
+  const [selectedProductIndex, setSelectedProductIndex] = useState(null);
+
+  const fetchItems = async () => {
+    try {
+      
+      const response = await api.get("/items");
+      
+      if (!response.ok) {
+        throw new Error(`HTTP error! Status: ${response.status}`);
       }
-      break
-    case "Running Low":
-      style = {
-        width: "90px",
-        backgroundColor: "white",
-        color: "orange",
-        padding: "5px",
-        borderRadius: "20px",
-        textAlign: "center",
-        border: "3px , solid ,orange",
+      
+      const data = await response.json();
+      console.log("Fetched data:", data);
+      setProductData(Array.isArray(data) ? data : []);
+    } catch (error) {
+      console.error("Error fetching items:", error);
+      setProductData([]);
+    }
+  };
+
+  const createItem = async (item) => {
+    try {
+      const response = await api.post("/items", item);
```

Comment:
Correct route should be `/api/items`. You probably should add a token as well as this is a admin only route.

```js
// get the token
const {token} = useAuth()

// create the request
const response = await api.post("/api/items", item, token)
```